File: 0015-3sum/0015-3sum.py
class Solution:
    def threeSum(self, nums: list[int]) -> list[list[int]]:
        nums.sort()
        n = len(nums)
        ans = set()
        for i in range(n-2):
            low = i
            mid = i+1
            high = n-1
            while(mid < high):
                a = nums[low]
                b = nums[mid]
                c = nums[high]
                total = a + b + c
                if total == 0:
                    ans.add(tuple([a,b,c]))
                    mid = mid + 1
                elif total > 0:
                    high = high-1
                else:
                    mid = mid + 1
        return [i for i in ans]

File: 0015-3sum/0015-3sum.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        set<vector<int>> myset;
        int n =nums.size();
        vector<vector<int>> final;
        sort(nums.begin(),nums.end());
        for (int i=0; i<n ; i++){
            int low=i+1, high=n-1;
            while (low<high){
                int sum=nums[low]+nums[high]+nums[i];
                if (sum==0){
                    myset.insert({nums[i],nums[low],nums[high]});
                    low++,high--;
                }
                else if(sum<0){
                    low++;
                }else high--;
            }
        }

        vector<vector<int>> ans(myset.begin(),myset.end());
        return ans;
        
    }
};